src/servers/Scripts/quest/classquest/ARC/ClsArc000.cpp

```diff
@@ -0,0 +1,180 @@
+#include <Actor/Player.h>
+#include <Event/EventHelper.h>
+#include <ScriptObject.h>
+
+// Quest Script: ClsArc000_00021
+// Quest Name: Way of the Archer
+// Quest ID: 65557
+// Start NPC: 1000197
+// End NPC: 1000200
+
+class ClsArc000 : public EventScript
+{
+   private:
+      // Basic quest information 
+      // Quest vars / flags used
+      // GetQuestUI8AL
+      // GetQuestUI8BH
+      // GetQuestUI8BL
+
+      enum Sequence : uint8_t
+      {
+         Seq0 = 0,
+         Seq1 = 1,
+         Seq2 = 2,
+         SeqFinish = 255,
+      };
+
+      // Quest rewards 
+      static constexpr auto RewardExpFactor = 100;
+      static constexpr auto RewardItem = 4657;
+      static constexpr auto RewardItemCount = 5;
+
+      // Entities found in the script data of the quest
+      static constexpr auto Actor0 = 1000197;
+      static constexpr auto Actor1 = 1000200;
+      static constexpr auto Enemy0 = 37;
+      static constexpr auto Enemy1 = 49;
+      static constexpr auto Enemy2 = 47;
+      static constexpr auto LocFace0 = 604;
+      static constexpr auto LocFace1 = 605;
+      static constexpr auto LogmessageMonsterNotePageUnlock = 1008;
+      static constexpr auto Seq0Actor0 = 0;
+      static constexpr auto Seq1Actor1 = 1;
+      static constexpr auto Seq3Actor1 = 2;
+      static constexpr auto UnlockImageMonsterNote = 32;
+
+   public:
+      ClsArc000() : EventScript( 65557 )
+      {}; 
+      ~ClsArc000()
+      {}; 
+
+   void onTalk( uint32_t eventId, Entity::Player& player, uint64_t actorId ) override
+   {
+      auto actor = Event::mapEventActorToRealActor( static_cast< uint32_t >( actorId ) );
+
+      if( actor == Actor0 )
+      {
+         Scene00000( player );
+      }
+      if( actor == Actor1 && player.getQuestSeq( getId() ) == Seq1 )
+      {
+         Scene00001( player );
+      }
+      if( actor == Actor1 && player.getQuestSeq( getId() ) == SeqFinish )
+      {
+         Scene00002( player );
+      }
+   }
+
+   void onMobKill( Entity::Player& player, uint64_t npcId )
+   {
+      if( npcId != Enemy0 || Enemy1 || Enemy2 )
```

Comment:
i assume you actually want to check if npcId is != any of those, in that case you have to write 
if( npcId != Enemy0 || npcId != Enemy1 || npcId != Enemy2 )

Reply:
The code you suggested isn't correct and would always be true it should be :
if( npcId != Enemy0 && npcId != Enemy1 && npcId != Enemy2 )
         return;